reolink-docs: added note as to why some sensors don't update (#27119)
home-assistant/home-assistant.github.io: source/_integrations/reolink.markdown

- Older firmware versions do not expose the necessary information the integration needs to function. Ensure the camera is updated to the [latest firmware](https://reolink.com/download-center/) prior to setting up the integration. Note that Reolink auto update and check for update functions in the app/windows/web client often do not show the latest available firmware version. Therefore check the version in the [Reolink download center](https://reolink.com/download-center/) online.
- Ensure at least one of the HTTP/HTTPS ports is enabled in the [windows](https://reolink.com/software-and-manual/)/web client under `Settings`->`Network`->`Advanced`->`Port Settings`, see [additional instructions](https://support.reolink.com/hc/en-us/articles/900004435763-How-to-Set-up-Reolink-Ports-Settings-via-Reolink-Client-New-Client-) on the Reolink site.
- On some camera models, the RTMP port needs to be enabled in order for the HTTP(S) port to function properly. Make sure this port is also enabled if you get a `Cannot connect to host` error while one of the HTTP/HTTPS ports is already enabled.

> - There is a known issue with the firmware of Reolink cameras not supporting webhooks behind a HTTPS endpoint. Home Assistant instances that are only available to your camera over HTTPS (i.e. both the internal and external interface of Home Assistant are configured with certificates and use HTTPS) then certain webhooks will not work. For example, the Visitor binary sensor from the Video Doorbell will not be updated when the doorbell button is pressed.

Reviewer: In the suggestion:

- It is not an issue but a limitation. Documenting it as an issue may seem like it is unresolvable or should be reported.

```suggestion
- Reolink cameras not supporting webhooks behind an HTTPS endpoint. Home Assistant instances that are only available to your camera over HTTPS (for example, both the internal and external interface of Home Assistant are configured with certificates and use HTTPS), then certain webhooks will not work. For example, the visitor binary sensor from the Video Doorbell will not be updated when the doorbell button is pressed.
```

I do still have a concern with this line, as it is unclear/ambiguous. I think you are trying to explain this is an issue when the Home Assistant instance is configured to use SSL in the `http` integration?

Also, what is the action a user has to take?

Author: There are work arounds, but they are in the realm of advanced/expert users and should certainly not be documented  here :-)

I wasted quite a bit of time trying to track down what was going on before I finally found an issue and PR in the core repo. I'm trying to stop others wasting their time.

Maybe instead of it being under a troubleshooting section, there should be a "Known Limitations" section and document that "Due to a camera firmware issue some cameras are not being able to communicate with HTTPS endpoints and as such some sensors will not update when the camera sends an event via a webhook".

There is already a section that mentions E1 Pro and E1 cameras having issues, could something be added there?

Reviewer: > There are work arounds, but they are in the realm of advanced/expert users and should certainly not be documented here :-)

Is it? The upcoming repairs issue will instruct users to use a reverse proxy by using an add-on.

Author: And that's great to hear there is going to a fix in the next release, as mentioned in the other conversation this is a _current_ issue that the PR is aimed it. :-)

Reviewer: Sure, I got that, my last comment did not say anything like that?

Author: How do you wish to progress? I've put forward a PR to help others out with an issue which is current. As best I can tell there is nothing that can be done in the current release to resolve this without telling users to edit code and set up a reverse proxy.

Short of getting Reolink to do something at the firmware side of things I don't know how to progress this now. Please provide a suggestion.

Reviewer 2: As frenck mentioned, HomeAssistant will offer repair issues with instructions on how to configure HomeAssistant diffrently (Use HTTP internally and use a reverse proxy addon for HTTPS externally) in order to get Reolink working properly in case a situation is detected in which the Reolink integration can not properly function.

Therefore personnally I don't think a extra warning needs to be present in the documentation, since that might confuse people that do not have any issues at all.

If we add this, we schould not say "certain webhooks will not work" but explicitly state what will and what will not work: everything will work except for the binary sensors: motion, people, face, vehicle, pet and visitor (detection).

Reviewer 2: @craiggenner maybe you could review PR https://github.com/home-assistant/core/pull/91597 since that will be the last missing peace to inform users about network configuration issues.

> - As of April 2023 the Reolink cameras do not support sending webhooks to a HTTPS endpoint. HA instances that are only available to your camera over HTTPS (i.e. both the internal and external interface of HA are configured with certificates and use HTTPS) then certain webhooks will not work. For example, the `Visitor` binary sensor from the Video Doorbell will not be updated when the doorbell button is pressed. There is no sanctioned solution to this, see the [github issue](https://github.com/home-assistant/core/issues/91640) for more details.

Reviewer: - "As of April 2023", did it work before that? I'm confused about why a time is added here.
- We don't track issues in docs, please remove the issue reference.
- Don't abbreviate HA, that is not needed in this sentence.
- "Visitor" is not code, and should not be code fenced.
- 2023.5 (beta start this week), will have detection for this and inform users in the repair center. Not sure if this is needed to be added because of that.

Author: > * "As of April 2023", did it work before that? I'm confused about why a time is added here.

Removed, I added so that readers have an idea of when this issue is know and how old is it.

> * We don't track issues in docs, please remove the issue reference.

Removed as requested

> * Don't abbreviate HA, that is not needed in this sentence.

Expanded

> * "Visitor" is not code, and should not be code fenced.

No longer quoted 

> * 2023.5 (beta start this week), will have detection for this and inform users in the repair center. Not sure if this is needed to be added because of that.

What happens if this doesn't make the beta (or final release)? Or if someone if still running an old version. Isn't it better to at least note the issue than waste time having them raise an issue.

Reviewer: > What happens if this doesn't make the beta (or final release)?

That is not a concern of this PR.

> Or if someone if still running an old version

Or documentation `current` branch only reflects the latest stable version. Others are not a consideration or concern.

Author: That's my point, I'm trying to document a current issue!

It may or may not be an issue in the next release - but it is _currently_ an issue. :-)
